packages/shared/src/react/hooks/useReverification.ts

```diff
@@ -28,8 +29,21 @@ async function resolveResult<T>(
     });
 }
 
-function createReverificationHandler(params: { onOpenModal: Clerk['__experimental_openUserVerification'] }) {
-  function assertReverification<Fetcher extends () => Promise<any>>(fetcher: Fetcher): Fetcher {
+type ExcludeClerkError<T, P> = T extends { clerk_error: any } ? (P extends { throwOnCancel: true } ? never : null) : T;
+
+type UseReverificationOptions = {
+  onCancel?: () => void;
+  throwOnCancel?: boolean;
+};
+
+function createReverificationHandler(
+  params: { onOpenModal: Clerk['__experimental_openUserVerification'] } & UseReverificationOptions,
```

Comment:
Can we include `onOpenModal` in the above options type?

Reply:
this should not be configurable from the developers, maybe if i rename to `openModal` is more clear. Basically we do `onOpenModal: __experimental_openUserVerification,` which is an internal concern.